[UnifiedPDF] Upstream enablement and platform availability information

aprotyas opened the pull request:
#### c81353986424c2561d715ecdf9eaa2aad1cf95f9
<pre>
[UnifiedPDF] Upstream enablement and platform availability information
<a href="https://bugs.webkit.org/show_bug.cgi?id=277014">https://bugs.webkit.org/show_bug.cgi?id=277014</a>
<a href="https://rdar.apple.com/132413786">rdar://132413786</a>

Reviewed by Wenson Hsieh and Aditya Keerthi.

This commit migrates feature enablement and platform availability
information related to UnifiedPDF from WebKitAdditions.

We take this opportunity to make a couple of clean ups:

- Rename ENABLE_UNIFIED_PDF_BY_DEFAULT to ENABLE_UNIFIED_PDF_AS_PREVIEW
  since the former was a gross misnomer. This flag is only used to define
  the feature status.

- Move around our declaration of the `-[PDFAction nextActions]` SPI in a
  way that makes said declaration visible to open source builds.

- Build UnifiedPDFTests for all platforms where UnifiedPDF is enabled,
  not only those where it is enabled by default. Also, configure the
  test web views in the respective tests appropriately by enabling the
  UnifiedPDF feature.

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WTF/wtf/PlatformHave.h:
* Source/WebKit/Platform/spi/Cocoa/PDFKitSPI.h:
* Source/WebKit/Shared/WebPreferencesDefaultValues.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/UnifiedPDFTests.mm:
(TestWebKitAPI::configurationForWebViewTestingUnifiedPDF):
(TestWebKitAPI::TEST(UnifiedPDF, KeyboardScrollingInSinglePageMode)):
(TestWebKitAPI::TEST(UnifiedPDF, CopyEditingCommandOnEmptySelectionShouldNotCrash)):

Canonical link: <a href="https://commits.webkit.org/281334@main">https://commits.webkit.org/281334@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/413461642dd24d45cf6dcc3d32d0480ddc6346dd

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/59552 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/38897 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/12076 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/63467 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/10073 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/61681 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/46550 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/10227 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/48322 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/7056 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/61582 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/36312 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/51553 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/29157 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/33016 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/8805 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/8999 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/52644 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/54963 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/9083 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/65199 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/58797 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/3480 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/8973 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/55662 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/3491 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/51545 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/55782 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/2890 "Passed tests") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/80554 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8895 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/34711 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/13970 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/35794 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/36880 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/35539 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Source/WebKit/Platform/spi/Cocoa/PDFKitSPI.h:
```diff
@@ -179,6 +171,14 @@ typedef NS_ENUM(NSUInteger, PDFSelectionGranularity);
+#if HAVE(PDFKIT_WITH_NEXT_ACTIONS) && __has_include(<PDFKit/PDFActionPriv.h>)
+#import <PDFKit/PDFActionPriv.h>
+#else
+@interface PDFAction (Staging_132413786)
```
~Hm... I would've expected 132413786 to be a bug in PDFKit.~

Oh, I see, this isn't new SPI on `PDFAction`. We should get this from the SPI header when using the internal SDK, and otherwise forward declare it in a `PDFActionPriv` category for non-internal builds.

aprotyas replied:
Right, of course. I'll change this to rdar://122820963

aprotyas replied:
Although, now that I think about it, currently this is less about staging the change and more about having the declaration available for public SDK builds. Maybe I'll just rename the category to ~~`SPI`~~`PDFActionPriv`.

aprotyas committed:
[UnifiedPDF] Upstream enablement and platform availability information
https://bugs.webkit.org/show_bug.cgi?id=277014
rdar://132413786

Reviewed by Wenson Hsieh and Aditya Keerthi.

This commit migrates feature enablement and platform availability
information related to UnifiedPDF from WebKitAdditions.

We take this opportunity to make a couple of clean ups:

- Rename ENABLE_UNIFIED_PDF_BY_DEFAULT to ENABLE_UNIFIED_PDF_AS_PREVIEW
  since the former was a gross misnomer. This flag is only used to define
  the feature status.

- Move around our declaration of the `-[PDFAction nextActions]` SPI in a
  way that makes said declaration visible to open source builds.

- Build UnifiedPDFTests for all platforms where UnifiedPDF is enabled,
  not only those where it is enabled by default. Also, configure the
  test web views in the respective tests appropriately by enabling the
  UnifiedPDF feature.

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WTF/wtf/PlatformHave.h:
* Source/WebKit/Platform/spi/Cocoa/PDFKitSPI.h:
* Source/WebKit/Shared/WebPreferencesDefaultValues.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/UnifiedPDFTests.mm:
(TestWebKitAPI::configurationForWebViewTestingUnifiedPDF):
(TestWebKitAPI::TEST(UnifiedPDF, KeyboardScrollingInSinglePageMode)):
(TestWebKitAPI::TEST(UnifiedPDF, CopyEditingCommandOnEmptySelectionShouldNotCrash)):

Canonical link: https://commits.webkit.org/281334@main
